Client/mods/deathmatch/logic/CClientVehicle.cpp

```diff
@@ -1061,8 +1089,34 @@ void CClientVehicle::SetModelBlocking(unsigned short usModel, unsigned char ucVa
         m_ucMaxPassengers = CClientVehicleManager::GetMaxPassengerCount(usModel);
 
         // Reset handling to fit the vehicle
-        m_pOriginalHandlingEntry = g_pGame->GetHandlingManager()->GetOriginalHandlingData((eVehicleTypes)usModel);
+        ushort usHandlingModelID = usModel;
+        if (usHandlingModelID < 400 || usHandlingModelID > 611)
+            usHandlingModelID = m_pModelInfo->GetParentID();
+
+        m_pOriginalHandlingEntry = g_pGame->GetHandlingManager()->GetOriginalHandlingData((eVehicleTypes)usHandlingModelID);
         m_pHandlingEntry->Assign(m_pOriginalHandlingEntry);
+
+        m_pOriginalFlyingHandlingEntry = g_pGame->GetHandlingManager()->GetOriginalFlyingHandlingData((eVehicleTypes)usHandlingModelID);
+        m_pFlyingHandlingEntry->Assign(m_pOriginalFlyingHandlingEntry);
+
+        m_pOriginalBoatHandlingEntry = g_pGame->GetHandlingManager()->GetOriginalBoatHandlingData((eVehicleTypes)usHandlingModelID);
+        if (m_pOriginalBoatHandlingEntry)
+        {
+            if (!m_pBoatHandlingEntry)
+                m_pBoatHandlingEntry = g_pGame->GetHandlingManager()->CreateBoatHandlingData();
+
+             m_pBoatHandlingEntry->Assign(m_pOriginalBoatHandlingEntry);
+        }
+
+        m_pOriginalBikeHandlingEntry = g_pGame->GetHandlingManager()->GetOriginalBikeHandlingData((eVehicleTypes)usHandlingModelID);
+        if (m_pOriginalBikeHandlingEntry)
+        {
+            if (!m_pBikeHandlingEntry)
+                m_pBikeHandlingEntry = g_pGame->GetHandlingManager()->CreateBikeHandlingData();
+
+            m_pBikeHandlingEntry->Assign(m_pOriginalBikeHandlingEntry);
```

Comment:
I dont understand why the original code has used this, when you have `operator=`.
So this would become `*m_pBikeHandlingEntry = *m_pOriginalBikeHandlingEntry)

Reply:
Because `*m_pBikeHandlingEntry = *m_pOriginalBikeHandlingEntry)` will copy `m_bDeleteInterface` field. I'm not sure that it will works without `CBikeHandlingEntrySA` declaration in `sdk/game_sa`